activestorage/test/service/s3_service_test.rb

```diff
@@ -0,0 +1,52 @@
+require "service/shared_service_tests"
+require "httparty"
+
+if SERVICE_CONFIGURATIONS[:s3]
+  class ActiveStorage::Service::S3ServiceTest < ActiveSupport::TestCase
+    SERVICE = ActiveStorage::Service.configure(:s3, SERVICE_CONFIGURATIONS)
+
+    include ActiveStorage::Service::SharedServiceTests
+
+    test "direct upload" do
+      begin
+        key      = SecureRandom.base58(24)
+        data     = "Something else entirely!"
+        checksum = Digest::MD5.base64digest(data)
+        url      = @service.url_for_direct_upload(key, expires_in: 5.minutes, content_type: "text/plain", content_length: data.size, checksum: checksum)
+
+        HTTParty.put(
+          url,
+          body: data,
+          headers: { "Content-Type" => "text/plain", "Content-MD5" => checksum },
+          debug_output: STDOUT
+        )
+
+        assert_equal data, @service.download(key)
+      ensure
+        @service.delete key
+      end
+    end
+
+    test "signed URL generation" do
+      assert_match /#{SERVICE_CONFIGURATIONS[:s3][:bucket]}\.s3.(\S+)?amazonaws.com.*response-content-disposition=inline.*avatar\.png.*response-content-type=image%2Fpng/,
```

Comment:
See my comment above https://github.com/rails/rails/pull/30020#discussion_r130663525

This test fails in the same way.